modules/puppet/manifests/init.pp

```diff
@@ -12,18 +12,31 @@
 #   Boolean, whether or not to use Puppet's future parser to
 #   help upgrade to Puppet 4.
 #
+# [*gem_bootstrap*]
+#   Set to true if the initial install of Puppet on the target
+#   system was via RubyGems. This will remove the gem installed
+#   Puppet as well as installing the Debian packages.
+#
 class puppet (
     $future_parser = false,
     $use_puppetmaster = true,
+    $gem_bootstrap = false,
   ) {
 
   validate_bool($future_parser)
   validate_bool($use_puppetmaster)
 
-  include puppet::cronjob
-  include puppet::repository
-  include puppet::package
-  include puppet::monitoring
+  include '::puppet::cronjob'
+  contain '::puppet::cronjob'
+
```

Comment:
I'm not sure I understand the difference between `include` and `contain` - [this](https://puppet.com/docs/puppet/5.5/lang_containment.html#containing-classes) doesn't help, yet includes both examples. This is a more fundamental question than probably appropriate for this code review, though.

Reply:
Containment means that if a resource forms an ordering relationship with the containing class, it also forms the same ordering relationship with the contained class.

From another look at the docs it looks like the `include` might be redundant in this case.. I'll try taking that out.